test/unit/specs/Calendar.spec.js

```diff
@@ -1,42 +1,328 @@
+        }
+      }
+
+      vm.$store.state.allSlides = [mockSlide]
+
+      Vue.nextTick(() => {
+        expect(vm.events.length).to.equal(1)
+        expect(vm.events[0].start).to.equal(mockSlide.date.content)
+        done()
+      })
+    })
+
+    it('should have correct start given no time content', function (done) {
+      let mockSlide = {
+        title: {
+          content: 'The Last One'
+        },
+        date: {
+          content: '2018-02-15'
+        },
+        time: {
+          fontColor: 'Red',
+          fontSize: 'Large',
+          fontStyle: 'Normal',
+          fontWeight: 'Bold'
+        }
+      }
+
+      vm.$store.state.allSlides = [mockSlide]
+
+      Vue.nextTick(() => {
+        expect(vm.events.length).to.equal(1)
+        expect(vm.events[0].start).to.equal(mockSlide.date.content)
+        done()
+      })
+    })
+
+    it('should be empty given empty slide', function (done) {
+      vm.$store.state.allSlides = [{}]
+
+      Vue.nextTick(() => {
+        expect(vm.events.length).to.equal(1)
+        expect(isEmptyObject(vm.events[0])).to.equal(true)
+        done()
+      })
+    })
+
+    it('should be empty given empty title', function (done) {
+      vm.$store.state.allSlides = [{
+        title: {},
+        date: {
+          content: '2018-02-15'
+        }
+      }]
+
+      Vue.nextTick(() => {
+        expect(vm.events.length).to.equal(1)
+        expect(isEmptyObject(vm.events[0])).to.equal(true)
+        done()
+      })
+    })
+
+    it('should be empty given no title content property', function (done) {
+      vm.$store.state.allSlides = [{
+        title: {
+          fontColor: 'Red',
+          fontSize: 'Large',
+          fontStyle: 'Normal',
+          fontWeight: 'Bold'
+        },
+        date: {
+          content: '2018-02-15'
+        }
+      }]
+
+      Vue.nextTick(() => {
+        expect(vm.events.length).to.equal(1)
+        expect(isEmptyObject(vm.events[0])).to.equal(true)
+        done()
+      })
+    })
+
+    it('should be empty given empty date', function (done) {
+      vm.$store.state.allSlides = [{
+        title: {
+          content: 'The Last One'
+        },
+        date: {}
+      }]
+
+      Vue.nextTick(() => {
+        expect(vm.events.length).to.equal(1)
+        expect(isEmptyObject(vm.events[0])).to.equal(true)
+        done()
+      })
+    })
+
+    it('should be empty given no date content property', function (done) {
+      vm.$store.state.allSlides = [{
+        title: {
+          content: 'The Last One'
+        },
+        date: {
+          fontColor: 'Red',
+          fontSize: 'Large',
+          fontStyle: 'Normal',
+          fontWeight: 'Bold'
+        }
+      }]
+
+      Vue.nextTick(() => {
+        expect(vm.events.length).to.equal(1)
+        expect(isEmptyObject(vm.events[0])).to.equal(true)
+        done()
+      })
     })
   })
+
+  // checks if obj is an empty Object i.e. obj = {}
+  function isEmptyObject (obj) {
```

Comment:
I believe that using `expect(obj).to.deep.equal({})` will be sufficient instead of creating a separate method just for checking empty object

Reply:
Yeah I forgot about that, fixed now.